src/anomalib/models/components/base/export_mixin.py

```diff
@@ -213,18 +220,34 @@ def to_openvino(
         if not try_import("openvino"):
             logger.exception("Could not find OpenVINO. Please check OpenVINO installation.")
             raise ModuleNotFoundError
+        if not try_import("nncf"):
+            logger.exception("Could not find NNCF. Please check NNCF installation.")
+            raise ModuleNotFoundError
 
+        import nncf
         import openvino as ov
 
         with TemporaryDirectory() as onnx_directory:
             model_path = self.to_onnx(onnx_directory, input_size, transform, task)
             export_root = _create_export_root(export_root, ExportType.OPENVINO)
             ov_model_path = export_root / "model.xml"
             ov_args = {} if ov_args is None else ov_args
-            # fp16 compression is enabled by default
-            compress_to_fp16 = ov_args.get("compress_to_fp16", True)
 
             model = ov.convert_model(model_path, **ov_args)
+            if compression_type == CompressionType.INT8:
+                model = nncf.compress_weights(model)
+            elif compression_type == CompressionType.INT8_PTQ:
+                assert datamodule is not None, "datamodule must be provided for OpenVINO INT8_PTQ compression"
+                dataloader = datamodule.val_dataloader()
+                if len(dataloader.dataset) < 300:
+                    logger.warning(
+                        f">300 images recommended for INT8 quantization, found only {len(dataloader.dataset)} images",
+                    )
```

Comment:
Is this mandatory. Not sure if all anomaly detection datasets have more than 300 validation images.

Reply:
Not mandatory, but recommended. That;s why it's warning only :)